File: das_backend/src/main/java/ch/sbb/backend/admin/infrastructure/entities/ServicePointEntity.java
package ch.sbb.backend.admin.infrastructure.entities;

import ch.sbb.backend.admin.domain.ServicePoint;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity(name = "service_point")
public class ServicePointEntity {

    @Id
    Integer uic;

    String designation;

    String abbreviation;

    public ServicePointEntity(ServicePoint sp) {
        this.uic = sp.uic();
        this.designation = sp.designation();
        this.abbreviation = sp.abbreviation();
    }

    public ServicePointEntity() {
        
    }

    public ServicePoint toServicePoint() {
        return new ServicePoint(uic, designation, abbreviation);
    }

}

File: backend/src/main/kotlin/ch/sbb/backend/servicepoints/infrastructure/entities/ServicePointEntity.kt
package ch.sbb.backend.servicepoints.infrastructure.entities

import ch.sbb.backend.servicepoints.domain.ServicePoint
import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity(name = "service_points")
class ServicePointEntity(
    @Id
    var uic: Int,
    var designation: String,
    var abbreviation: String
) {


    constructor(servicePoint: ServicePoint) : this(
        servicePoint.uic,
        servicePoint.designation,
        servicePoint.abbreviation
    )

    fun toServicePoint():ServicePoint {
       return ServicePoint(uic,designation, abbreviation)
    }
}
